### PR TITLE
[CA-1776] Expose `GET` workspaces/{namespace}/{name}/migrate

### PR DESCRIPTION
**Pull request by ehigham**
RR: https://broadworkbench.atlassian.net/browse/CA-1776
Add endpoint to return a list of workspace migration attempt metadata.
These objects represent a simplified view of a migration attempt.
Add json support to utility classes.

Spec: https://broadworkbench.atlassian.net/wiki/spaces/CI/pages/2358149130/CA-1776+GET+Migration+Status

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Review by gpolumbo-broad — APPROVED** [review]

File: core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiService.scala
Comment (andy7i) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiService.scala]: the API being a `GET  ... migrate` is a bit weird to me. 

I wonder if there's something else we can do? Maybe renaming `migrate` to `migration`?

```suggestion
      path("workspaces" / Segment / Segment / "migration") { (namespace, name) =>
```

then it works with GET and PUT

**Review by andy7i — COMMENTED** [review]
I'm assuming that nobody (or at least nobody improper) has the permission for `SamWorkspaceActions.migrate`

I didn't see any swagger documentation for the migrate APIs. Do we want to add it to swagger or keep the APIs unlisted?

**Comment by ehigham** [comment]
> I'm assuming that nobody (or at least nobody improper) has the permission for `SamWorkspaceActions.migrate`
> 
> I didn't see any swagger documentation for the migrate APIs. Do we want to add it to swagger or keep the APIs unlisted?

Yeah, I have not included swagger doc deliberately as I want this api to be undocumented - as far as use cases are concerned, we're just using it for tests and we'll delete it when all workspaces are migrated. If there's a use case to expose it to users then we can consider that, I just don't want to commit to maintaining an API that's temporary in nature.

File: core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiService.scala
Comment (ehigham) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiService.scala]: I agree in part but `PUT` migration seems a bit strange too. `POST` might be more appropriate in that case but is `POST` with an empty body ok?

Comment (andy7i) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiService.scala]: throwing out there that we could go with a more verbose option by creating an endpoint for `GET ... migrationStatus`.
I think since this is an undocumented API, sticking to one endpoint might make things easier for us. I think I still prefer `migration` over `migrate`, but I also don't feel super strongly about it.

**Review by andy7i — CHANGES_REQUESTED** [review]
I just took a look at https://github.com/broadinstitute/sam/blob/develop/src/main/resources/reference.conf#L130 and it looks like we grant `migrate` to all workspace owners. That doesn't seem right and I don't think we should expose this API until we figure this out

File: core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiService.scala
Comment (ehigham) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiService.scala]: I changed to POST and renamed to `migrations`

**Comment by ehigham** [comment]
> I just took a look at https://github.com/broadinstitute/sam/blob/develop/src/main/resources/reference.conf#L130 and it looks like we grant `migrate` to all workspace owners. That doesn't seem right and I don't think we should expose this API until we figure this out

Why does that not seem right? Why shouldn't anyone who can create a workspace migration attempt get these attempts?

**Comment by ehigham** [comment]
> > I just took a look at https://github.com/broadinstitute/sam/blob/develop/src/main/resources/reference.conf#L130 and it looks like we grant `migrate` to all workspace owners. That doesn't seem right and I don't think we should expose this API until we figure this out
> 
> Why does that not seem right? Why shouldn't anyone who can create a workspace migration attempt get these attempts?

As discussed in the mob - our plan is to remove `migrate` from workspace owners for the time being and grant it to admins so we can test the actor. I'll create a PR in sam for this. No further changes on this subject are required here.

**Review by andy7i — APPROVED** [review]
(After some discussion:) Looks like the PUT is already out there so it doesn't make sense to block this PR due to the permission. 

Let's create a ticket to figure out the right way we want "feature flag" this, but that shouldn't block this PR.

Side note: Did we get AppSec's blessing for removing the feature flag on these endpoints?

**Comment by ehigham** [comment]
> (After some discussion:) Looks like the PUT is already out there so it doesn't make sense to block this PR due to the permission.
> 
> Let's create a ticket to figure out the right way we want "feature flag" this, but that shouldn't block this PR.

Created https://broadworkbench.atlassian.net/browse/CA-1782

> Side note: Did we get AppSec's blessing for removing the feature flag on these endpoints?

Removing the feature flag?

**Pull request merged**
